docs/samples/Microsoft.ML.Samples/Static/LightGBMMulticlassWithInMemoryData.cs

```diff
@@ -0,0 +1,163 @@
+                        Predictions: ctx.Trainers.LightGbm(r.Label.ToKey(), r.Features)
+                    ))
+                    .Append(r => (
+                        // Actual label.
+                        r.Label,
+                        // Labels are converted to keys when training LightGBM so we convert it here again for calling evaluation function.
+                        LabelIndex: r.Label.ToKey(),
+                        // Used to compute metrics such as accuracy.
+                        r.Predictions,
+                        // Assign a new name to predicted class index.
+                        PredictedLabelIndex: r.Predictions.predictedLabel,
+                        // Assign a new name to class probabilities.
+                        Scores: r.Predictions.score
+                    ));
+
+            // Split the static-typed data into training and test sets. Only training set is used in fitting
+            // the created pipeline. Metrics are computed on the test.
+            var (trainingData, testingData) = ctx.TrainTestSplit(staticDataView, testFraction: 0.5);
+
+            // Train the model.
+            var model = pipe.Fit(trainingData);
+
+            // Do prediction on the test set.
+            var prediction = model.Transform(testingData);
+
+            // Evaluate the trained model is the test set.
+            var metrics = ctx.Evaluate(prediction, r => r.LabelIndex, r => r.Predictions);
+
+            // Check if metrics are resonable.
+            Console.WriteLine ("Macro accuracy: {0}, Micro accuracy: {1}.", 0.863482146891263, 0.86309523809523814);
+
+            // Convert prediction in ML.NET format to native C# class.
+            var nativePredictions = new List<NativeExample>(prediction.AsDynamic.AsEnumerable<NativeExample>(mlContext, false));
+
+            // Get schema object out of the prediction. It contains metadata such as the mapping from predicted label index
+            // (e.g., 1) to its actual label (e.g., "AA"). The call to "AsDynamic" converts our statically-typed pipeline into
+            // a dynamically-typed one only for extracting metadata. In the future, metadata in statically-typed pipeline should
+            // be accessible without dynamically-typed things.
+            var schema = prediction.AsDynamic.Schema;
+
+            // Retrieve the mapping from labels to label indexes.
+            var labelBuffer = new VBuffer<ReadOnlyMemory<char>>(); 
+            schema[nameof(NativeExample.PredictedLabelIndex)].Metadata.GetValue("KeyValues", ref labelBuffer);
+            var nativeLabels = labelBuffer.DenseValues().ToList(); // nativeLabels[nativePrediction.PredictedLabelIndex-1] is the original label indexed by nativePrediction.PredictedLabelIndex.
+
+            // Show prediction result for the 3rd example.
+            var nativePrediction = nativePredictions[2];
+            Console.WriteLine("Our predicted label to this example is {0} with probability {1}",
```

Comment:
> Console.WriteLine(" [](start = 11, length = 20)

have a comment for the WriteLines showing here directly what the output would  be.

Reply:
No problem. Will do the same for Line 159 below.

---
In reply to: [244876086](https://github.com/dotnet/machinelearning/pull/1953#discussion_r244876086) [](ancestors = 244876086)